sys/xtimer/xtimer.c

```diff
@@ -217,7 +223,36 @@ int xtimer_msg_receive_timeout(msg_t *msg, uint32_t us)
 {
     msg_t tmsg;
     xtimer_t t;
+
     _setup_timer_msg(&tmsg, &t);
     xtimer_set_msg(&t, us, &tmsg, sched_active_pid);
     return _msg_wait(msg, &tmsg, &t);
 }
+
+static void _mutex_timeout(void *arg)
+{
+    unsigned irqstate = irq_disable();
```

Comment:
@lebrush RIOT currently doesn't allow nested interrupts, so while one ISR is running, no other will interrupt.